### ISSUE TITLE
Two questions

### ISSUE THREAD

**Issue by 675492062** [AndreyBocharnikov/Visual-Transformers#1]
Thank you for your efforts. 

1) For semantic segmentation, the output of 'semanticsessionbranch' uses softmax, and then nn.crossentropyloss is used when computing the final loss. Are you sure?

2) Why you ignores the label 163? It is so weird.

Looking forward to your reply! Thanks for advance.

**Comment by AndreyBocharnikov** [comment]
Hello! Sorry for long reply, haven't received any notification about your question :(
1. I think you are right, from the nn.crossentropyloss documentation "The input is expected to contain raw, unnormalized scores for each class", so I guess softmax should be deleted and output should be transposed appropriately.
2. It's because the logic [here](https://github.com/AndreyBocharnikov/Visual-Transformers/blob/master/data/semantic_segmentation.py#L78), I skip first 91 classes because it things (or stuff, I don't remember) I all first 91 classes are equals to the 255 class (163 in my case) which is void or unlabelled class as I remember.
3. Also I thing I've got a bug [here](https://github.com/AndreyBocharnikov/Visual-Transformers/blob/master/models/semantic_segmentation.py#L14), it should be for loop instead of multiplication, because there are shadow copies of one deepcopy that is in the array.

I'm sorry I won't be maintaining this repo, but fill free to ask any questions, I will be glad to help you.

**Comment by 675492062** [comment]
> Hello! Sorry for long reply, haven't received any notification about your question :(
> 
> 1. I think you are right, from the nn.crossentropyloss documentation "The input is expected to contain raw, unnormalized scores for each class", so I guess softmax should be deleted and output should be transposed appropriately.
> 2. It's because the logic [here](https://github.com/AndreyBocharnikov/Visual-Transformers/blob/master/data/semantic_segmentation.py#L78), I skip first 91 classes because it things (or stuff, I don't remember) I all first 91 classes are equals to the 255 class (163 in my case) which is void or unlabelled class as I remember.
> 3. Also I thing I've got a bug [here](https://github.com/AndreyBocharnikov/Visual-Transformers/blob/master/models/semantic_segmentation.py#L14), it should be for loop instead of multiplication, because there are shadow copies of one deepcopy that is in the array.
> 
> I'm sorry I won't be maintaining this repo, but fill free to ask any questions, I will be glad to help you.

Thank you for your kindly reminding and pointing, especially the third point.
